### PR TITLE
Add up_to_date_checker.js

### PR DESCRIPTION
**Pull request by xinbinhuang**
This component will check if the PR is up to date with master only if certain files specified in the config are modified as discussed in the mail thread.

### CODE REVIEW

**Comment by xinbinhuang** [comment]
@potiuk @kaxil  Please take a look.

The config will look something like this. https://github.com/xinbinhuang/airflow/commit/cd2b0863bbd5424d08d6425b5aeb9bbb0034ff2d

And this is an example PR https://github.com/xinbinhuang/airflow/pull/7

**Comment by kaxil** [comment]
> @potiuk @kaxil Please take a look.
> 
> The config will look something like this. [xinbinhuang/airflow@cd2b086](https://github.com/xinbinhuang/airflow/commit/cd2b0863bbd5424d08d6425b5aeb9bbb0034ff2d)
> 
> And this is an example PR [xinbinhuang/airflow#7](https://github.com/xinbinhuang/airflow/pull/7)

Awesome, thanks for the PR @xinbinhuang . Reviewing it now

File: lib/up_to_date_checker.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
+'use strict'
+const ignore = require('ignore')
+
+/**
+ * Check branch up to date with master if specific files are modified in the PR
```
Comment (kaxil) [inline lib/up_to_date_checker.js]:
```suggestion
 * Check branch is up to date with master if specific files are modified in the PR
```

Comment (xinbinhuang) [reply lib/up_to_date_checker.js]: I rephrased the docstring. I hope it's more readable right now.

Comment (xinbinhuang) [reply lib/up_to_date_checker.js]: I rephrased the docstring. I hope it's more readable right now.

**Comment by xinbinhuang** [comment]
> If possible, can you add some docs around this awesome feature @xinbinhuang in https://github.com/kaxil/boring-cyborg/blob/master/README.md#features
> 
> and Config in https://github.com/kaxil/boring-cyborg/blob/master/README.md#usage

Added^. Please take a look. After this is merged, I will also submit a PR to the bot setting. Do you think the config example is good?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "boring-cyborg",
-  "version": "1.0.0",
+  "version": "1.1.0",
```
Comment (kaxil) [inline package.json]:
```suggestion
  "version": "1.0.0",
```

**Comment by kaxil** [comment]
> > If possible, can you add some docs around this awesome feature @xinbinhuang in https://github.com/kaxil/boring-cyborg/blob/master/README.md#features
> > and Config in https://github.com/kaxil/boring-cyborg/blob/master/README.md#usage
> 
> Added^. Please take a look. After this is merged, I will also submit a PR to the bot setting. Do you think the config example is good?

Yup, it is definitely good. Well done @xinbinhuang 🎉

**Comment by xinbinhuang** [comment]
Thanks @kaxil !

**Pull request merged**